### DISCUSSION TITLE
Intermittent "Invalid API key" error

### DISCUSSION THREAD

**Discussion by frankbarry** [etsy/open-api#1026] [Q&A]
Hi - I get an occasional "Invalid API key" error from the Etsy api. When I retry a few minutes later, the call succeeds. This isn't a huge blocker but it is annoying. Is this happening because of something I'm doing wrong, or is it just an intermittent blip on the Etsy side?

**Comment by cmunden** [comment]
Can you provide the full details of both the request and the response with the error?  

Intermittent failures that work a few minutes later sound like an issue with throttling and call limits.   I've been using the new API quite a bit over the last 2 weeks and have never every had an issue with invalid api keys.

**Reply by frankbarry** [reply]
I've been using the api for about 6 months without issues, this started happening recently, 3 times this month.

These were all POSTs to `https://openapi.etsy.com/v3/application/shops/{shopId}/receipts/{receiptId}/tracking`. In one case, a subsequent call to the same API completed successfully, less than 1 second later. 

The failed call was preceded by a POST to `https://api.etsy.com/v3/public/oauth/token` to acquire a refresh token.

Throttling and call limits do not fit. These are just a few calls, and the error message doesn't match. I suspect it is related to the refresh token not being valid immediately after acquisition in certain situations, perhaps a race condition on the etsy side, or maybe on my side.
